tests/security/tpm2/tpm2_env_setup.pm

```diff
@@ -42,22 +46,17 @@ ConditionPathExistsGlob=
 EOF
         )\" > /etc/systemd/system/tpm2-abrmd.service.d/emulator.conf"
     );
-    assert_script_run "systemctl daemon-reload";
-    assert_script_run "systemctl enable tpm2-abrmd";
-
-    # Reboot the node to make the changes take effect
-    power_action('reboot', textmode => 1);
-    $self->wait_boot(textmode => 1);
-    $self->select_serial_terminal;
+    assert_script_run("systemctl daemon-reload");
+    assert_script_run("systemctl enable tpm2-abrmd");
 
     # Start the emulator
-    my $server = script_output('ls /usr/*/ibmtss/tpm_server | tail -1');
-    die 'missing tpm_server path\n' if ($server eq '');
-    assert_script_run "su - tss -c '$server &'";
+    my $server = script_output("ls /usr/*/ibmtss/tpm_server | tail -1");
+    die "missing tpm_server path\n" if ($server eq '');
+    assert_script_run("su - tss -c '$server &'");
```

Comment:
You won't need the `usermod` above if you use

```suggestion
    assert_script_run("su -s /bin/bash - tss -c '$server &'");
```

Or even

```
    background_script_run("su -s /bin/sh - tss -c '$server'");
```

BTW, what stops the process at the end again?

Reply:
Cool, let me try

I didn't stop the process, since many chained tests need connect the emulate server now.

Reply:
I may have to use 'usermod', tpm2-abrmd service need tss account as well.
```
See "systemctl status tpm2-abrmd.service" and "journalctl -xeu tpm2-abrmd.service" for details.
```

Reply:
> I may have to use 'usermod', tpm2-abrmd service need tss account as well.

Yes, but it already exists. The usermod only sets a shell. If the service unit is broken without doing system modifications, that would be a product bug.

Reply:
I don't think it is a product bug, since we used emulated ibm tpm2 server rather than a real tpm device.

Reply:
We have done some workaround to create a user 'tss' to start tpm server and tpm2-abrmd service at first :)

Reply:
The emulated tpm2 server is part of the product, so any issues with it are product bugs.

Reply:
Let me double check my code then

Reply:
Well, found the RC, background_script_run seems take some time to start the tpm server, and tpm2-abrmd failed to start if it can't communicate with tpm server.

Reply:
Hm, that can happen with the old code as well though... Maybe that was just more lucky.

Reply:
Yes, agree with you! so far so good with old code.